src/main/java/CMV.java

```diff
@@ -260,7 +260,42 @@ private boolean LIC8() {
    * Computes the LIC 9 condition.
    */
   private boolean LIC9() {
-    // TODO: Implementation.
+
+    // NUMPOINTS >= 5
+    if (points.length < 5) {
+      return false;
+    }
+
+    // C_PTS >= 1, D_PTS >= 1
+    if (parameters.C_PTS < 1 || parameters.D_PTS < 1) {
+      return false;
+    }
+
+    // C_PTS + D_PTS <= NUMPOINTS - 3
+    if (parameters.C_PTS + parameters.D_PTS > points.length - 3) {
+      return false;
+    }
+
+
+    for (int i = 0; i < points.length - parameters.C_PTS - parameters.D_PTS - 2; i++) {
+      var a = points[i];
+      var vert = points[i + parameters.C_PTS + 1];
+      var b = points[i + parameters.C_PTS + 1 + parameters.D_PTS + 1];
+
+      // Points coincide at vertex, angle undefined, does not satisfy LIC
+      if (a.equals(vert) || b.equals(vert)) {
+        continue;
+      }
```

Comment:
Technically nothing wrong here; could perhaps be a bit more clear to do a `return false` immediately, I was a bit confused at first :)

Reply:
This just means those three points in particular don't satisfy the LIC, but there are more points to check, hence continue.

Reply:
That's very true, I guess I mixed up "at least one" with "all" (since we could immediately return 'false' in the latter case)